preset/ilib.js

```diff
@@ -1,33 +1,55 @@
+import {memoize} from '@enact/core/util';
+import ilib from '@enact/i18n';
 import ResBundle from '@enact/i18n/ilib/lib/ResBundle';
 
 import {configure as conf} from '../index';
 
+const memoMapper = memoize((/* locale */) => {
+	// Retrieve the ResBundle to get access to the string map for the current locale.
+	const resBundle = new ResBundle();
+	const map = resBundle.getResObj();
+
+	// We have to convert the case since the text may be converted by UI components to a
+	// different case during render.
+	const reversed = Object.keys(map).reduce((acc, key) => {
+		acc[map[key].toUpperCase()] = key;
+		return acc;
+	}, {});
+
+	return (value) => reversed[value.toUpperCase()];
+});
+
+const getMapper = () => memoMapper(ilib.getLocale());
+
 const config = {
 	format: (msg) => {
-		// Retrieve the ResBundle to get access to the string map for the current locale.
-		const resBundle = new ResBundle();
-		const map = resBundle.getResObj();
+		const findValue = getMapper();
 
 		// For every key (except type and time), attempt to delocalize the string.
-		Object.keys(msg).filter(key => key !== 'type' && key !== 'time').forEach(msgKey => {
-			// We have to convert the case since the text may be converted by UI components to a
-			// different case during render.
-			const upperValue = msg[msgKey].toUpperCase();
-			const value = Object.keys(map).find(key => upperValue === map[key].toUpperCase());
-
-			if (value) {
-				msg[msgKey] = value;
-			}
-		});
+		Object.keys(msg)
+			.filter(key => key !== 'type' && typeof msg[key] === 'string')
```

Comment:
Is `'time'` not a filtered key any more?  If not, should we update the comment above?

Reply:
It's filtered by its `typeof` but i'll update the comment to reflect that.

Reply:
Is this a critical path?  Would it make sense to dispense with the `.filter()` step and do it inline with the `forEach()`?  Too much micro-opt?

Reply:
Seems like a micro opt to me. In my 3 key test, this method is only taking 0.01ms. Assuming it's linear, there'd have to be a lot of keys in the message payload to make a meaningful impact.

Reply:
My only thought was along the lines of reducing pressure on GC.  But, as you say, it's probably not likely they are large arrays being allocated/deallocated.